projects/igniteui-angular/src/lib/directives/forms-control/forms-control.directive.ts

```diff
@@ -0,0 +1,99 @@
+import {
+    Directive,
+    forwardRef,
+    ElementRef,
+    HostListener,
+    NgModule,
+    Renderer2,
+    AfterViewInit,
+    OnDestroy
+} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { fromEvent, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Directive({
+    selector: 'igc-rating', /* eslint-disable-line @angular-eslint/directive-selector */
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            useExisting: forwardRef(() => IgxFormsControlDirective),
+            multi: true
+        }
+    ]
+})
+export class IgxFormsControlDirective implements ControlValueAccessor, AfterViewInit, OnDestroy {
+    /** @hidden @internal */
+    public onChange: any = () => { };
+    /** @hidden @internal */
+    public onTouched: any = () => { };
+
+    protected _destroy$ = new Subject();
+
+    private _value: any;
+
+    /** @hidden @internal */
+    public get value(): any {
+        return this.elementRef.nativeElement.value;
+    }
+    public set value(val) {
```

Comment:
Not entirely sold on the purpose of the value getter/setter. Other than the change callback scenario in tests that is, and even there not really required. The purpose is to enable form-control functionality, right? In that case you're dealing with `ngModel` prop or a `FormControl` for reading/writing values, right?

PS: The hidden/internal comment I'm fairly certain applies to the getter only, so you end up with a public setter, not sure if intended